lib/jekyll/readers/layout_reader.rb

```diff
@@ -26,8 +28,10 @@ def layout_directory
       @layout_directory ||= site.in_source_dir(site.config["layouts_dir"])
     end
 
-    def theme_layout_directory
-      @theme_layout_directory ||= site.theme.layouts_path if site.theme
+    def theme_layout_directory(theme)
```

Comment:
You shouldn't change the signature of a `public` method. Please define a new method to wrap around the existing one.

Reply:
@ashmaroli is that the only thing preventing us from shipping this feature? Can we help here to move this forward?

Reply:
Sounds good. I've updated the PR to add a new method `theme_layout_directory_for_theme(theme)` instead.